src/vehicle.cpp

```diff
@@ -4758,21 +4787,25 @@ vehicle_part_range vehicle::get_parts() const
 }
 
 template<>
-bool vehicle_part_with_feature_range<std::string>::contained( const size_t part ) const
+bool vehicle_part_with_feature_range<std::string>::matches( const size_t part,
+        const part_status_flag e ) const
 {
     const vehicle_part &vp = this->vehicle().parts[part];
     return vp.info().has_flag( feature_ ) &&
            !vp.removed &&
-           ( !unbroken_ || !vp.is_broken() ) &&
-           ( !enabled_ || vp.enabled );
+           ( !( e && part_status_flag::working ) || !vp.is_broken() ) &&
+           ( !( e && part_status_flag::available ) || vp.is_available() ) &&
+           ( !( e && part_status_flag::enabled ) || vp.enabled );
```

Comment:
The `&&` is boolean-and, the condition is equivalent `e != 0 and part_status_flag::working != 0`, which is equivalent to `e != 0` (as the second operand is always `true`). Did you mean `&` (arithmetic operator)?

Reply:
Yes.

Reply:
Fixed.